pom.xml

```diff
@@ -887,7 +887,7 @@ under the License.
 			<dependency>
 				<groupId>io.netty</groupId>
 				<artifactId>netty-bom</artifactId>
-				<version>4.1.70.Final</version>
+				<version>4.1.100.Final</version>
```

Comment:
Hi, @snuyanzin  Thank you for the netty version update. I have been working with netty recently and have noticed a hardcoded netty version in the flink-rpc-akka/pom.xml (https://github.com/apache/flink/blob/b759794e33d68b557f4ee2d7287f5fae9a80167e/flink-rpc/flink-rpc-akka/pom.xml#L98). Do we need to update that version? 
In addition, it appears that the original hardcoded version might not be accurate.

Reply:
Hi @TanYuxin-tyx 
thanks for the question
the reason why upgrade of netty and arrow is required is failing of number of python tests while building with jdk21

I have my  own pipepline setup with jdk21
example of error could be seen e.g. https://dev.azure.com/snuyanzin/flink/_build/results?buildId=2490&view=logs&j=fba17979-6d2e-591d-72f1-97cf42797c11&t=727942b6-6137-54f7-1ef9-e66e706ea068

same error was mentioned in FLINK-33331 description.

Since both netty and arrow require this removed constructor during python test they both need bumping the version.
By the way here it is the result of same pipeline with cherry-picked commit from this PR https://dev.azure.com/snuyanzin/flink/_build/results?buildId=2491&view=results
it could be seen that after that python tests with java 21 are passing.


About the dependency you've mentioned: this dependency is used **only** for Pekko (former Akka) since current Pekko version can not run with netty 4.x... At the same time based on tests it is not an issue, looks like they don't used this removed constructor...

Reply:
One more note about flink-rpc-akka dependency

in theory it could be update in future within a separate task
however first there should be support of Netty 4 from Pekko's side which is done within https://github.com/apache/incubator-pekko/pull/643
however it is not released yet...

Reply:
@snuyanzin OK, make sense. Thanks for your patient explanations.